TFEngine.py

```diff
@@ -1950,7 +1991,13 @@ def extra_fetches_callback(seq_idx, seq_tag, output, targets=None, beam_scores=N
       assert len(out_cache) - 1 in out_cache
       if output_file_format == "txt":
         for i in range(len(out_cache)):
-          output_file.write("%s\n" % out_cache[i])
+          if isinstance(out_cache[i], dict):
+            # manually construct dict string to guarantee deterministic order
+            out_string = "{" + ', '.join("%r: %s" % (output_layer_names[t], out_cache[i][output_layer_names[t]]) \
+                for t in range(num_serializable_targets)) + "}\n"
```

Comment:
There is `Util.betterRepr` for this purpose.
Also then you don't need the comment (as it becomes clear when you use `Util.betterRepr`).

Reply:
However, I think it makes more sense to just don't allow txt format output for this case anyway. That makes it just more complicated to parse it later. As said: Don't make it more complicated than necessary.

Reply:
Note that you should use `Util.betterRepr` for the py format case, though.